cmd/ef_tests/state/runner_v2/result_check.rs

```diff
@@ -1,63 +1,102 @@
-use std::{fs::OpenOptions, io::Write, path::PathBuf};
+use std::collections::{BTreeMap, HashMap};
 
-use ethrex_common::types::AccountUpdate;
+use bytes::Bytes;
+use ethrex_common::{
+    Address, U256,
+    types::{Account, AccountUpdate, Fork, Genesis},
+};
 use ethrex_levm::{
+    db::gen_db::GeneralizedDatabase,
     errors::{ExecutionReport, TxValidationError, VMError},
     vm::VM,
 };
+use ethrex_rlp::encode::RLPEncode;
 use ethrex_storage::Store;
 use ethrex_vm::backends;
-use keccak_hash::H256;
+use keccak_hash::{H256, keccak};
 
 use crate::runner_v2::{
     error::RunnerError,
-    types::{Test, TestCase, TransactionExpectedException},
+    types::{AccountState, TestCase, TransactionExpectedException},
 };
 
-/// Creates a basic `.txt` report that informs the result of executing each test (if it passed or failed).
-pub fn create_report(
-    res: Result<(), RunnerError>,
-    test_case: &TestCase,
-    test: &Test,
-) -> Result<(), RunnerError> {
-    let report_path = PathBuf::from("./runner_v2/runner_report.txt");
-    let mut report = OpenOptions::new()
-        .append(true)
-        .create(true)
-        .open(report_path)
-        .unwrap();
-    let content = if res.is_ok() {
-        format!(
-            "Test checks succeded for test: {:?}, with fork {:?},  in path: {:?}.\n",
-            test.name, test_case.fork, test.path
-        )
-    } else {
-        format!(
-            "Test checks failed for test: {:?}, with fork: {:?},  in path: {:?}.\n",
-            test.name, test_case.fork, test.path,
-        )
-    };
-    let _ = report.write_all(content.as_bytes());
-
-    Ok(())
+pub struct PostCheckResult {
+    pub fork: Fork,
+    pub vector: (usize, usize, usize),
+    pub passed: bool,
+    pub root_dif: Option<(H256, H256)>,
+    pub accounts_diff: Option<Vec<AccountMismatch>>,
+    pub logs_diff: Option<(H256, H256)>,
+    pub exception_diff: Option<(Vec<TransactionExpectedException>, Option<VMError>)>,
+}
+impl Default for PostCheckResult {
+    fn default() -> Self {
+        Self {
+            fork: Fork::Prague,
+            vector: (0, 0, 0),
+            passed: true,
+            root_dif: None,
+            accounts_diff: None,
+            logs_diff: None,
+            exception_diff: None,
+        }
+    }
+}
+#[derive(Debug, Clone)]
+pub struct AccountMismatch {
+    pub address: Address,
+    pub expected_balance: U256,
+    pub actual_balance: U256,
+    pub expected_nonce: u64,
+    pub actual_nonce: u64,
+    pub expected_code: Bytes,
+    pub actual_code: Bytes,
+    pub expected_storage: HashMap<U256, U256>,
+    pub actual_storage: BTreeMap<H256, U256>,
```

Comment:
Can we use the same type for both of these fields? The formatting for U256 and H256 is different, making reports weird. For example:
```
Expected storage: {0: 966588469268559010541288244128342317224451555083}
Actual   storage: {0x0000000000000000000000000000000000000000000000000000000000000000: 115792089237316195423570985008687907853269984665640564039457584007913129639935}
```
I also think we should keep the `BTreeMap` for both fields as it is easier to compare large diffs with sorted keys